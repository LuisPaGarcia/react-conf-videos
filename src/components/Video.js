import React from 'react'
import { connect } from 'react-redux'
import { compose, pure } from 'recompose'

const VideoInner = ({ video }) => {
  const { title, duration, link } = video
  return (
    <div>
      <a href="{link}">{title}</a> : {duration}
    </div>
  )
}

const mapStateToProps = (state, props) => {
  const { data: { videos } } = state
  const { videoId } = props
  return {
    video: videos[videoId]
  }
}

const Video = compose(
  connect(mapStateToProps, null),
  pure
)(VideoInner)

export default Video
