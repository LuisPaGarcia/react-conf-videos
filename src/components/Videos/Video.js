import React from 'react'
import { connect } from 'react-redux'
import { compose, pure, withStateHandlers } from 'recompose'
import * as cn from 'classnames'

import styles from './Video.scss'

const VideoInner = ({
  video, speaker, videoId, conference, isOpen, toggleIsOpen
}) => {
  const {
    title, length, link, embeddableLink
  } = video
  return (
    <div className={styles.root} key={videoId} >
      <div className={styles.top} onClick={toggleIsOpen} >
        <span className={styles.title} href={link}>{title}</span>
        <span className={styles.right}>{length}</span>
      </div>
      <div className={cn(styles.videoWrapper, { [styles.open]: isOpen })}>
        {isOpen && <iframe
          title="videoPlayer"
          id="ytplayer"
          type="text/html"
          width="410"
          height="360"
          src={embeddableLink}
          frameBorder="0"
          allowFullScreen="allowfullscreen"
          mozallowfullscreen="mozallowfullscreen"
          msallowfullscreen="msallowfullscreen"
          oallowfullscreen="oallowfullscreen"
          webkitallowfullscreen="webkitallowfullscreen"
        /> }
      </div>
      <div className={styles.details}>
        <span>{speaker.name}</span>
        <span>{conference.title}</span>
      </div>
    </div>
  )
}

const mapStateToProps = (state, props) => {
  const { data: { videos, presenters, conferences } } = state
  const { videoId, conferenceId } = props
  const video = videos[videoId]
  const conference = conferences[conferenceId]
  const speaker = presenters[video.presenter]
  return {
    video: videos[videoId],
    conference,
    speaker
  }
}

const Video = compose(
  connect(mapStateToProps, null),
  pure,
  withStateHandlers({ isOpen: false }, {
    toggleIsOpen: ({ isOpen }) => () => ({ isOpen: !isOpen })
  })
)(VideoInner)

export default Video
