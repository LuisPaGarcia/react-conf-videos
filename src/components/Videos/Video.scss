@import '../../styles/_colors';

.root {
  width: 100%;
  padding: 1rem 2rem;
  box-sizing: border-box;
  font-size: 1.6rem;
  color: $text-color;

  .top {
    display: flex;
    justify-content: space-between;

    .title {
      font-weight: bold;
    }

    .right {
      font-size: 1.6rem;
    }
  }

  .details {
    display: flex;
    justify-content: space-between;
    font-size: 1.4rem;

    span {
      font-size: 1.4rem;
    }    
  }

  .divider {
    height: 1px;
    width: 100%;
    background-color: grey;
  }

}