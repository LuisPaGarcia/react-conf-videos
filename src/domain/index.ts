import { Action } from './Action'
import { Video, IndexedVideos } from './Video'
import { Presenter, IndexedPresenters } from './Presenter'
import { Conference, IndexedConferences } from './Conference'
import { ConferenceInput, VideoInput, PresenterInput, JSONInput } from './InputJSON'

export {
  Action,
  Video, IndexedVideos,
  Presenter, IndexedPresenters,
  Conference, IndexedConferences,
  ConferenceInput, VideoInput, PresenterInput, JSONInput
}