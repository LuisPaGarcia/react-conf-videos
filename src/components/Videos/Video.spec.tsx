import * as React from 'react'
import { VideoInner } from './Video'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { Video, Conference } from '../../domain'

describe('Video', () => {
  const mockFN = jest.fn()

  const shallowVideo = () => {
    const video = {
      title: 'test title', 
      length: '12:34', 
      link: 'a link', 
      embeddableLink: 'a link'
    }
    const conferenceId = 'conferenceId1234'
    const conference: Partial<Conference> = { title: 'A conference' }
    const videoId = 'videoId1234'
    const speaker = { name: 'Simon Carter'}
    const isOpen = false
    const toggleIsOpen = mockFN

    return shallow<any>(
      <VideoInner 
        conferenceId={conferenceId}
        video={video as Video}
        videoId={videoId}
        speaker={speaker}
        isOpen={isOpen}
        toggleIsOpen={toggleIsOpen}
        conference={conference as Conference}
      />
    )
  }

  it('renders correctly', () => {
    // arrange
    const wrapper = shallowVideo()
    // act
    // assert
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('handles onClick', () => {
    // arrange
    const wrapper = shallowVideo()
    
    // act
    wrapper.find('.top').simulate('click')

    // assert
    expect(mockFN).toHaveBeenCalled()
  });

})

