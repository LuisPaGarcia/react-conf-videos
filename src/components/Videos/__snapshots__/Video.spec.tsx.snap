// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video renders correctly 1`] = `
<div
  className="root"
  key="videoId1234"
>
  <div
    className="top"
    onClick={[MockFunction]}
  >
    <a
      className="title"
      href="a link"
    >
      test title
    </a>
    <span
      className="right"
    >
      12:34
    </span>
  </div>
  <div
    className="videoWrapper"
  />
  <div
    className="details"
  >
    <span>
      Simon Carter
    </span>
    <span>
      A conference
    </span>
  </div>
</div>
`;
